.github/workflows/build.yml

```diff
@@ -110,3 +110,153 @@ jobs:
       with:
         name: Ubuntu DEB
         path: cmake_build/*.deb
+
+  windows-msvc:
+    name: Windows / MSVC
+    runs-on: windows-2019
+    steps:
+    - name: Check out repository
+      uses: actions/checkout@v2
+    - name: Set up cmake
+      uses: jwlawson/actions-setup-cmake@v1.4
+      with:
+        cmake-version: '3.13.x'
+    - name: Set up build environment
+      id: setup_environment
+      shell: cmake -P {0}
+      run: |
+        file(READ "build/windows/golden_environment" BUILDENV_NAME)
+        string(STRIP "${BUILDENV_NAME}" BUILDENV_NAME)
+        string(REPLACE "PLATFORM" "$ENV{PLATFORM}" BUILDENV_NAME "${BUILDENV_NAME}")
+        string(REPLACE "CONFIGURATION" "$ENV{CONFIGURATION}" BUILDENV_NAME "${BUILDENV_NAME}")
+
+        message("::group::Download environment")
+        set(BUILDENV_URL "$ENV{BUILDENV_BASEURL}/${BUILDENV_NAME}.zip")
+        set(BUILDENV_FILE "./${BUILDENV_NAME}.zip")
+        message("URL: ${BUILDENV_URL}")
+        file(DOWNLOAD "${BUILDENV_URL}" "${BUILDENV_FILE}" SHOW_PROGRESS STATUS DOWNLOAD_STATUS)
+        list(GET DOWNLOAD_STATUS_LIST 0 DOWNLAOD_STATUS)
+        if (NOT DOWNLOAD_STATUS EQUAL 0)
+            message(FATAL_ERROR "Downloading environment failed!")
+        endif()
+        message("::endgroup")
```

Comment:
This should be cached on GitHub Actions.

Reply:
Cache the unpacked directory because GitHub Actions automatically zstd compresses caches.